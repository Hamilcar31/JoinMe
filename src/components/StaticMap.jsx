import React from 'react';
import {ImageBackground, StyleSheet, View} from 'react-native';
import MapView from 'react-native-maps';
import window from '../constants/Layout.js';
import {LinearGradient} from "expo-linear-gradient";
import {rgba} from "polished";
import {Shadows, Strokes} from "../constants";

export default (props) => {

    const {
        location = 'Epitech Toulouse',
        style = styles.image
    } = props;

    let preview = 'https://maps.googleapis.com/maps/api/staticmap?center=' + encodeURI(location) +
        '&zoom=13&size=1800x1800&maptype=roadmap' +
        '&markers=color:blue%7C' + encodeURI(location) +
        '&key=AIzaSyAfv8IPCxhiURtrI8tDyQptGEVQoOl0G3c';

    return (
        <ImageBackground style={[styles.image, style]} source={{uri: preview}}>
            <LinearGradient
                end={[0, 1]}
                colors={['#FFFFFF00', '#FFFFFF00', '#FFFFFF11', '#FFFFFF77', '#FFFFFFFF']}
                style={{
                    width: '100%',
                    height: '100%'
                }}
            />
        </ImageBackground>
    );
}

const styles = StyleSheet.create({
    image: {
        height: 400,
        left: 0,
        right: 0,
        top: 0,
        bottom: 0
    },
});
