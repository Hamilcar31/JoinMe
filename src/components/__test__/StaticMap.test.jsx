import React from 'react';
import renderer from 'react-test-renderer';

import StaticMap from '../StaticMap.jsx';

describe('StaticMap component', () => {
  it('has location', () => {
    // given
    const staticMap = <StaticMap location="Epitech Paris"/>;

    // when
    const tree = renderer.create(staticMap).toJSON();

    // then
    expect(tree).toMatchSnapshot();
  });
});