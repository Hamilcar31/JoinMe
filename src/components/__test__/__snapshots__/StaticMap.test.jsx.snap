// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StaticMap component has location 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Array [
      Object {
        "bottom": 0,
        "height": 400,
        "left": 0,
        "right": 0,
        "top": 0,
      },
      Object {
        "bottom": 0,
        "height": 400,
        "left": 0,
        "right": 0,
        "top": 0,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "https://maps.googleapis.com/maps/api/staticmap?center=Epitech%20Paris&zoom=13&size=1800x1800&maptype=roadmap&markers=color:blue%7CEpitech%20Paris&key=AIzaSyAfv8IPCxhiURtrI8tDyQptGEVQoOl0G3c",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          16777215,
          16777215,
          301989887,
          2013265919,
          4294967295,
        ],
        "endPoint": Array [
          0,
          0,
        ],
        "locations": undefined,
        "startPoint": Array [
          0,
          0.5,
        ],
      }
    }
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  />
</View>
`;
